jlm/llvm/backend/rvsdg2jlm/rvsdg2jlm.cpp

```diff
@@ -427,17 +427,16 @@ convert_lambda_node(const rvsdg::Node & node, context & ctx)
 static inline void
 convert_phi_node(const rvsdg::Node & node, context & ctx)
 {
-  JLM_ASSERT(rvsdg::is<phi::operation>(&node));
-  auto phi = static_cast<const rvsdg::StructuralNode *>(&node);
-  auto subregion = phi->subregion(0);
+  auto & phi = dynamic_cast<const rvsdg::PhiNode &>(node);
```

Comment:
Rather `util::AssertedCast<>`, no? I do not want an exception thrown here. If we do not get a phi node in here, then something is wrong in our code, not in the user input.